packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -188,14 +208,41 @@ function parseGlobOptions(
 
   if (opts.as) opts.query = opts.as
 
-  return opts
+  if (opts.query && opts.query[0] !== '?') opts.query = `?${opts.query}`
+
+  // validate `import` option based on `query` option (`as` is already handled above)
+  if (!opts.as && opts.query) {
+    if (urlRE.test(opts.query)) {
+      errorIfImportIsNotDefault(opts.import, '?url', optsStartIndex)
+      opts.import = opts.import || 'default'
+    }
+    if (rawRE.test(opts.query)) {
+      errorIfImportIsNotDefault(opts.import, '?raw', optsStartIndex)
+      opts.import = opts.import || 'default'
+    }
+  }
```

Comment:
I feel it a bit too magic to change the default of `import` option based on `query` option. To me, `as` feels it might transform the output, but `query` only changes the query when imported.
That said, I think it'd be handy with this.

Reply:
That's true. I didn't think too much about it 🤔 I would be fine removing this and have the migration guide request adding `import: 'default'` too, especially given that this isn't typed anymore. Would you prefer that too?

Reply:
I removed the default `import: 'default'` configuration. This should make it more predictable.